services/wallet/api.go

```diff
@@ -853,8 +854,37 @@ func (api *API) GetWalletConnectDapps(ctx context.Context, validAtTimestamp int6
 	return walletconnect.GetActiveDapps(api.s.db, validAtTimestamp, testChains)
 }
 
-// signTypedDataV4 dApps use it to execute "eth_signTypedData_v4" requests
+func hashMessageForSigning(data []byte) types.Hash {
+	safeMsg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(data), string(data))
+	return crypto.Keccak256Hash([]byte(safeMsg))
+}
+
+// HashMessageForSigning is used for handling dApps requests for "personal_sign" and "eth_sign"
+// in a safe manner following the EIP-191 version 0x45 for signing on the client side.
+func (api *API) HashMessageForSigning(ctx context.Context, data []byte) types.Hash {
+	log.Debug("wallet.api.PreparePersonalMessage", "len(data)", len(data))
+	return hashMessageForSigning(data)
+}
+
+// SignMessageForDApps is used for handling dApps requests for "personal_sign" and "eth_sign" requests
+func (api *API) SignMessageForDApps(ctx context.Context, message types.HexBytes, address common.Address, password string) (string, error) {
```

Comment:
I am not a fan of naming functions like `SignMessageForDApps`, is it matter if the signing is for dapps or not.

Since we already have `SignMessage` and `HashMessageForSigning` functions exposed by this API you can use them on the client side to achieve the same or if you really want to have those two calls combined in this API you can name it `HashMessageAndSign`.

Reply:
Good point, I forgot that there is already a `wallet_SignMessage`; I will use that. Thanks.

Reply:
what about just having smth like SignMessageEIP191? Not sure if we need to expose the hash really.

Reply:
I'm now testing a mix of both recommendations use `HashMessageEIP191` for hashing and the existing `SignMessage` for signing